### PR TITLE
End of Chapter 11: Account Activation

### PR DESCRIPTION
**Pull request by khanhtrant12**
Account Activation

### CODE REVIEW

**Comment by khanhtrant12** [comment]
ready

File: app/assets/stylesheets/account_activations.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the AccountActivations controller here.
```
Comment (huongnguyenmta) [inline app/assets/stylesheets/account_activations.scss]: xóa file

File: app/models/user.rb [hunk 2]
```diff
@@ -39,10 +40,11 @@ def remember
     update_attribute(:remember_digest, User.digest(remember_token))
   end
 
-  def authenticated? remember_token
+  def authenticated? attribute, token
+    digest = send("#{attribute}_digest")
```
Comment (huongnguyenmta) [inline app/models/user.rb]: tìm hiểu method `send`
keyword: `Ruby meta-programming`

File: app/views/layouts/mailer.text.erb [hunk 3]
```diff
@@ -1 +1,5 @@
-<%= yield %>
+Hi <%= @user.name %>,
+
+Welcome to the Sample App! Click on the link below to activate your account:
```
Comment (huongnguyenmta) [inline app/views/layouts/mailer.text.erb]: I18n

File: app/views/layouts/mailer.text.erb [hunk 4]
```diff
@@ -1 +1,5 @@
...
+
+<%= edit_account_activation_url(@user.activation_token, email: @user.email) %>
```
Comment (huongnguyenmta) [inline app/views/layouts/mailer.text.erb]: end file

File: test/controllers/account_activations_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (huongnguyenmta) [inline test/controllers/account_activations_controller_test.rb]: bỏ qua phần test em ơi @@

**Comment by khanhtrant12** [comment]
ready

**Pull request merged**